ext/re2/extconf.rb

```diff
@@ -6,6 +6,134 @@
 
 require 'mkmf'
 
+PACKAGE_ROOT_DIR = File.expand_path(File.join(File.dirname(__FILE__), '..', '..'))
+
+REQUIRED_MINI_PORTILE_VERSION = "~> 2.8.2" # keep this version in sync with the one in the gemspec
+
+RE2_HELP_MESSAGE = <<~HELP
+  USAGE: ruby #{$0} [options]
+
+    Flags that are always valid:
+
+      --use-system-libraries
+      --enable-system-libraries
+          Use system libraries instead of building and using the packaged libraries. This is the default.
+
+      --disable-system-libraries
+          Use the packaged libraries, and ignore the system libraries. This overrides `--use-system-libraries`.
+
+    Flags only used when using system libraries:
+
+      Related to re2 library:
+
+        --with-re2-dir=DIRECTORY
+            Look for re2 headers and library in DIRECTORY.
+
+    Environment variables used:
+
+      CC
+          Use this path to invoke the compiler instead of `RbConfig::CONFIG['CC']`
+
+      CPPFLAGS
+          If this string is accepted by the C preprocessor, add it to the flags passed to the C preprocessor
+
+      CFLAGS
+          If this string is accepted by the compiler, add it to the flags passed to the compiler
+
+      LDFLAGS
+          If this string is accepted by the linker, add it to the flags passed to the linker
+
+      LIBS
+          Add this string to the flags passed to the linker
+HELP
+
+#
+#  utility functions
+#
+def config_system_libraries?
+  enable_config("system-libraries", true) do |_, default|
+    arg_config("--use-system-libraries", default)
```

Comment:
In Nokogiri, `--use-system-libraries` only exists for backwards compatibility, I don't think it's necessary to provide more than one commandline flag for this.

Reply:
Thanks for that context!